packages/features/ee/workflows/lib/reminders/smsReminderManager.ts

```diff
@@ -139,180 +139,140 @@ export const scheduleSMSReminder = async (args: ScheduleTextReminderArgs) => {
+                    replyTo: evt.organizer.email,
+                  }
+                : undefined,
+            });
+          } catch (error) {
+            log.error(`Error sending SMS with error ${error}`);
+          }
+        }
 
-          if (scheduledNotification?.sid) {
-            await prisma.workflowReminder.create({
-              data: {
-                bookingUid: uid,
-                workflowStepId: workflowStepId,
-                method: WorkflowMethods.SMS,
+        if (
+          (triggerEvent === WorkflowTriggerEvents.BEFORE_EVENT ||
+            triggerEvent === WorkflowTriggerEvents.AFTER_EVENT) &&
+          scheduledDate
+        ) {
+          try {
+            // schedule at least 15 minutes in advance and at most 2 hours in advance
+            const scheduledNotification = await scheduleSmsOrFallbackEmail({
+              twilioData: {
+                phoneNumber: reminderPhone,
+                body: smsMessage,
                 scheduledDate: scheduledDate.toDate(),
-                scheduled: true,
-                referenceId: scheduledNotification.sid,
-                seatReferenceId: seatReferenceUid,
+                sender: senderID,
+                bookingUid: evt.uid,
+                userId,
+                teamId,
               },
+              fallbackData: isAttendeeAction(action)
+                ? {
+                    email: evt.attendees[0].email,
+                    t: await getTranslation(evt.attendees[0].language.locale, "common"),
+                    replyTo: evt.organizer.email,
+                    workflowStepId,
+                  }
+                : undefined,
             });
+
+            if (scheduledNotification?.sid) {
+              await prisma.workflowReminder.create({
+                data: {
+                  bookingUid: uid,
+                  workflowStepId: workflowStepId,
+                  method: WorkflowMethods.SMS,
+                  scheduledDate: scheduledDate.toDate(),
+                  scheduled: true,
+                  referenceId: scheduledNotification.sid,
+                  seatReferenceId: seatReferenceUid,
+                },
+              });
+            }
+          } catch (error) {
+            log.error(`Error scheduling SMS with error ${error}`);
           }
-        } catch (error) {
-          log.error(`Error scheduling SMS with error ${error}`);
         }
-      } else if (scheduledDate.isAfter(currentDate.add(2, "hour"))) {
-        // Write to DB and send to CRON if scheduled reminder date is past 2 hours from now
-        await prisma.workflowReminder.create({
-          data: {
-            bookingUid: uid,
-            workflowStepId: workflowStepId,
-            method: WorkflowMethods.SMS,
-            scheduledDate: scheduledDate.toDate(),
-            scheduled: false,
-            seatReferenceId: seatReferenceUid,
-          },
-        });
       }
+      return;
+    }
+
+    if (!useTwilio && scheduledDate) {
```

Comment:
if `useTwilio` is false we don't need the sms message. It's generated in the CRON